src/schemas/Entity.js

```diff
@@ -52,12 +54,12 @@ export default class EntitySchema {
   }
 
   denormalize(entityOrId, unvisit, entities) {
-    const entity = typeof entityOrId === 'object' ? entityOrId : entities[this.key][entityOrId];
-    const entityCopy = { ...entity };
+    const entity = typeof entityOrId === 'object' ? entityOrId : getIn(entities, [ this.key, entityOrId ]);
+    let entityCopy = isImmutable(entity) ? entity : { ...entity };
     Object.keys(this.schema).forEach((key) => {
-      if (entityCopy.hasOwnProperty(key)) {
+      if (hasIn(entityCopy, [ key ])) {
```

Comment:
Can we use the `isImmutable(entity)`  method before doing `getIn`, `hasIn`, and `setIn` to avoid unnecessary overhead?